### PR TITLE
doc: Update sandbox to incubation

### PR DESCRIPTION
**Pull request by jwcesign**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
doc: Update sandbox to incubation

**Which issue(s) this PR fixes**:
Fixes #none

**Special notes for your reviewer**:
none

### CODE REVIEW

File: docs/core-concepts/introduction.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Karmada (Kubernetes Armada) is a Kubernetes management system that enables you t
 Karmada aims to provide turnkey automation for multi-cluster application management in multi-cloud and hybrid cloud scenarios,
 with key features such as centralized multi-cloud management, high availability, failure recovery, and traffic scheduling.
 
-Karmada is a sandbox project of the [Cloud Native Computing Foundation](https://cncf.io/) (CNCF).
+Karmada is an incubation project of the [Cloud Native Computing Foundation](https://cncf.io/) (CNCF).
```
Comment (zhzhuang-zju) [inline docs/core-concepts/introduction.md:14]: nice job~ there is a same statement in [karmada repo](https://github.com/karmada-io/karmada/blob/master/README.md?plain=1#L23) but mistakenly used 'an' to 'a', could you please modify it together? Thanks

Comment (jwcesign) [reply docs/core-concepts/introduction.md:14]: Got it

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**